### ISSUE TITLE
[BUG] linear activation and activation_hidden in SimpleRNNRegressor

### ISSUE THREAD

**Issue by RecreationalMath** [sktime/sktime#9047]
**Describe the bug**

RNN regressor currently has linear for both hidden and final layer activations, which essentially defeats the purpose of using a neural network and reduces the whole setup to linear regression.

**To Reproduce**

Check [line number 87 & 88 in SimpleRNNRegressor class](https://github.com/sktime/sktime/blob/main/sktime/regression/deep_learning/rnn.py#L87C1-L89C1).

**Expected behavior**

Standard practice is to use a non-linear activation so that complex patterns can be learned. 
Usually if the RNN is shallow, `tanh` is used, and if it is deep, `ReLU` or `leakyReLU` is used for better gradient propagation.

**Additional context**

If you use only linear activation functions in a deep neural network, all layers collapse into a single equivalent layer, rendering the network unable to learn complex, non-linear patterns. This is because any combination of linear functions is still a linear function, making the entire network behave like a single-layer linear model, such as linear regression. 
**Versions**
<details>

System:
    python: 3.12.10 (main, Oct 30 2025, 20:57:26) [Clang 17.0.0 (clang-1700.3.19.1)]
executable: /.pyenv/versions/sktime-dev-py-3.12.10/bin/python
   machine: macOS-15.6.1-arm64-arm-64bit

Python dependencies:
          pip: 25.3
       sktime: 0.39.0
      sklearn: 1.6.1
       skbase: 0.12.6
        numpy: 2.1.3
        scipy: 1.15.3
       pandas: 2.3.3
   matplotlib: 3.10.7
       joblib: 1.5.2
huggingface-hub: None
        numba: 0.62.1
     pmdarima: None
pytorch-forecasting: None
   skforecast: 0.18.0
        skpro: 2.9.4
statsforecast: 2.0.3
  statsmodels: 0.14.5
 transformers: None
      tsfresh: None
      tslearn: None
        torch: 2.9.0
   tensorflow: 2.19.1
</details>
